2.overview_and_terminology.md

```diff
@@ -20,38 +20,57 @@ This specification proposes a model that defines cloud native applications as fo
 
 > A cloud native application is a collection of interrelated, but discrete _components_ (services, tasks, workers) that, when coupled with configuration and instantiated in a suitable runtime, together accomplish a unified functional purpose.
 
-The application model defines the following:
+This specification defines the following application constructs for application developers and application operators to design and operate applications:
 
- - _Components_ represents a runnable unit, together with a description (schematic).
- - _Workload types_ identify the different workloads that a component can execute.
- - _Traits_ are overlays that augment a component with additional operations-specific features. Traits represent operator concerns, not developer concerns.
- - _Application scopes_ represent application boundaries by grouping components with common properties or dependencies.
- - An _application configuration_ describes a set of component instances, their traits, and the application scopes in which they are placed, combined with configuration parameters and metadata.
+- __Component__, where developers define an instance of a workload and declare the operational characteristics of the instance they deliver _in infrastructure neutral terms_.
+- __Traits__ for assigning operational features to instances of components.
+- __Application scopes__, a way to group components into loosely coupled applications with common characteristics.
+- __Application configuration__, defines a deployment of components, their traits, and application scopes.
 
-Thus, an _application_ is a collection of _components_ with a set of operational traits and scoped together into one or more application boundaries. 
+In addition, the specification defines the following definition constructs for infrastructure operators and platform builders to define the components, traits, and scopes that are available to application developers and application operators on a given platform:
+
+- __Workload Definition__, where platforms declare the schema of a specific kind of workload, such as containers, functions, or data stores, that developers use to create components.
+- __Trait Definition__, where platforms declare the schema of a specific kind of trait.
+- __Scope Definition__, where platforms declare the schema of a specific kind of application scope.
+
+Thus, OAM defines an _application_ as a collection of _components_ each with a set of operational traits and scoped together into one or more application boundaries.
```

Comment:
Can we have a diagram here? See the one I created below, it might help:

![image](https://user-images.githubusercontent.com/130954/77213165-1ca78c00-6ac7-11ea-9582-fcbf463d93bc.png)

Reply:
I think a diagram is a good idea. The suggested diagram mostly makes sense, but is not completely accurate. I would say:

Definitions:

* A trait definition defines a kind of trait - this is accurate in the diagram.
* A workload definition defines a kind of workload. A `kind: WorkloadDefinition` defines `kind: ContainerizedWorkload`.
* A scope definition defines a kind of scope. A `kind: ScopeDefinition` defines `kind: NetworkScope`.

Types

* An application scope is not a type of scope definition. A scope definition defines a type of application scope.
* It may be clearer to think of a component as _templating_ a type of workload. A component is not a type of workload.

Implementations

* `NetworkScope`, `ContainerizedWorkload`, and `ManualScalerTrait` implement scope, workload, and trait (which are concepts, or duck types). Not scope definition, workload definition, and trait definition.

Reply:
![image](https://user-images.githubusercontent.com/130954/77215088-917ec400-6acf-11ea-8745-8920762df2fb.png)

Reply:
Thanks, Artur. While a picture worth a thousand words, I am having some problems understanding this diagram.  Just to name a few

- I don't know what does the right side (yellowish part) of the graph represents
- I am not sure if I will say appConfig "config" a scope or a trait
- I am also not sure if a component is  "a type of " workloadDefinition. Maybe it is a type of workload?
- I am also not sure if a containerizedWorkload "implements" a workloadDefinition.

Reply:
Thanks for your feedback. Let me answer your questions:

1. Yellow, means it is an implementation of OAM and not OAM spec itself.
2. It seems that "config" was the most natural verb for relationships for "AppConfig" - any ideas of which verbs would be more appropriate? Maybe "injects Traits" and "defines Scopes" ?
3. Diagram has been updated since that comment. Please, check latest version above.
4. Same. Diagram is updated.